utils/defaulter.go

```diff
@@ -277,6 +301,28 @@ func (d *Defaulter) populateDefaultsFromInput(defaults interface{}) error {
 	return nil
 }
 
+func validateKongDefaults(defaults interface{}) error {
+	var errs ErrArray
+	r := reflect.ValueOf(defaults)
+	for objectName, restrictedFields := range defaultsRestrictedFields {
+		objectValue := reflect.Indirect(r).FieldByName(strings.Title(objectName))
+		if objectValue.IsNil() || !objectValue.IsValid() {
+			continue
+		}
+		object := objectValue.Interface()
+		err := checkEntityDefaults(object, restrictedFields)
+		if err != nil {
+			entityErr := fmt.Errorf(
+				"%s defaults %s", objectName, err)
+			errs.Errors = append(errs.Errors, entityErr)
+		}
+	}
+	if errs.Errors != nil {
+		return errs
+	}
+	return nil
+}
```

Comment:
I feel there is a little bit too much of reflection, but that's mainly caused by https://github.com/Kong/deck/blob/main/utils/defaulter.go#L27 (which I'm not sure why it is an interface{} and not a normal struct like in the file builder).

Reply:
Changing [this]((https://github.com/Kong/deck/blob/main/utils/defaulter.go#L27) would be a bit more involving, so if you feel this can benefit from a bit of refactoring turning it into a proper struct (I agree), please open an GH issue for tracking and let's tackle it separately!